src/helpers.js

```diff
@@ -52,30 +52,31 @@ export function getReportingParams($scope) {
 
 export function getNextPlaybackItemInfo() {
 
-    var playlist = window.playlist;
+    var playlist = window.playbackManager.activePlaylist;
 
     if (!playlist) {
         return null;
     }
 
+    let activeIndex = window.playbackManager.activePlaylistIndex;
     var newIndex;
 
-    if (window.currentPlaylistIndex == -1) {
+    if (activeIndex == -1) {
         newIndex = 0;
     } else {
         switch (window.repeatMode) {
 
             case 'RepeatOne':
-                newIndex = window.currentPlaylistIndex;
+                newIndex = activeIndex;
                 break;
             case 'RepeatAll':
-                newIndex = window.currentPlaylistIndex + 1;
-                if (newIndex >= window.playlist.length) {
+                newIndex = activeIndex + 1;
+                if (newIndex >= playlist.length) {
                     newIndex = 0;
                 }
                 break;
             default:
-                newIndex = window.currentPlaylistIndex + 1;
+                newIndex = activeIndex;
```

Comment:
I didn't notice any anomalies from removing the `+ 1` during my fairly extensive testing. I believe this was done to prevent the index from being -1 but those cases seems to be taken care of by this if statement above:

```js
if (activeIndex == -1) {
    newIndex = 0;
} else {
```

Reply:
This looks like it was done to increment the song during playback, otherwise it will always repeat the same song. The code for that could be in another location though, I haven't checked.

Reply:
That would make a lot of sense. All the queue stuff that was implemented has since stopped working (if it did post-CAF migration, can't quite remember) so I didn't test that scenario